### DISCUSSION TITLE
How to select the data items under the corresponding message in custom panel , just like the message path in the Gauge panel!!!!!!

### DISCUSSION THREAD

**Discussion by Kezunlong1025** [foxglove/community#1033] [Q&A]
How to select the data items under the corresponding message in custom panel , just like the message path in the Gauge panel!!!!!!
I'm desperate to get an answer to this question. Please help me!
thanks!
like this:
![PixPin_2024-09-05_17-26-11](https://github.com/user-attachments/assets/5354625f-694d-4299-a7a4-96a432f3fa71)

**Comment by defunctzombie** [comment]
I recommend reading the docs on message path syntax here: https://docs.foxglove.dev/docs/visualization/message-path-syntax/

If that is not enough, then please elaborate on what you mean by "exact information"? What specific part of your message do you want to surface? Your post does not describe which part of the message you want to identify.

**Reply by Kezunlong1025** [reply]
Thank you very much for taking the time to answer my questions!!!
The exact information I'm talking about is getting the exact item in the message path like the Message Path in the gauge panel.
Normally we can only select the message path, but not the exact item on our custom panel.

**Reply by Kezunlong1025** [reply]
I am developing an extension panel, and I want the panel to be able to select the data items under the corresponding message, just like the message path in the Gauge panel, rather than just selecting the path and stopping there.
What should I do? Looking forward to your reply!!

**Reply by jtbandes** [reply]
Sorry, we don't provide a utility for this right now — but we will discuss and consider possible solutions.

**Reply by Kezunlong1025** [reply]
> Sorry, we don't provide a utility for this right now — but we will discuss and consider possible solutions.
Well T-T , still very much appreciate your reply! Wishing you a pleasant life :) .
